Rubberduck.Parsing/VBA/DeclarationResolveRunnerBase.cs

```diff
@@ -20,33 +20,20 @@ public abstract class DeclarationResolveRunnerBase : IDeclarationResolveRunner
         protected readonly IParserStateManager _parserStateManager;
         private readonly IProjectReferencesProvider _projectReferencesProvider;
 
-        public DeclarationResolveRunnerBase(
+        protected DeclarationResolveRunnerBase(
             RubberduckParserState state,
             IParserStateManager parserStateManager,
             IProjectReferencesProvider projectReferencesProvider)
         {
-            if (state == null)
-            {
-                throw new ArgumentNullException(nameof(state));
-            }
-            if (parserStateManager == null)
-            {
-                throw new ArgumentNullException(nameof(parserStateManager));
-            }
-            if (projectReferencesProvider == null)
-            {
-                throw new ArgumentNullException(nameof(projectReferencesProvider));
-            }
+            _state = state ?? throw new ArgumentNullException(nameof(state));
```

Comment:
again: ArgumentNullExceptions are not to be simplified :wink:

Reply:
Overeagerness bites me. Good to know, working reversions.